src/ImageProcessor/Imaging/ResizeLayer.cs

```diff
@@ -102,67 +103,49 @@ public class ResizeLayer
         /// Gets or sets the anchor point.
         /// </summary>
         public Point? AnchorPoint { get; set; }
 
-        /// <summary>
-        /// Returns a value that indicates whether the specified object is an 
-        /// <see cref="ResizeLayer"/> object that is equivalent to 
-        /// this <see cref="ResizeLayer"/> object.
-        /// </summary>
-        /// <param name="obj">
-        /// The object to test.
-        /// </param>
+        /// <summary>Determines whether the specified object is equal to the current object.</summary>
+        /// <param name="obj">The object to compare with the current object.</param>
         /// <returns>
-        /// True if the given object  is an <see cref="ResizeLayer"/> object that is equivalent to 
-        /// this <see cref="ResizeLayer"/> object; otherwise, false.
-        /// </returns>
+        ///   <see langword="true" /> if the specified object  is equal to the current object; otherwise, <see langword="false" />.</returns>
         public override bool Equals(object obj)
-        {
-            if (!(obj is ResizeLayer resizeLayer))
-            {
-                return false;
-            }
+            => this.Equals(obj as ResizeLayer);
 
-            return this.Size == resizeLayer.Size
-                && this.ResizeMode == resizeLayer.ResizeMode
-                && this.AnchorPosition == resizeLayer.AnchorPosition
-                && this.Upscale == resizeLayer.Upscale
-                && ((this.CenterCoordinates != null
-                    && resizeLayer.CenterCoordinates != null
-                    && this.CenterCoordinates.SequenceEqual(resizeLayer.CenterCoordinates))
-                    || (this.CenterCoordinates == resizeLayer.CenterCoordinates))
-                && this.MaxSize == resizeLayer.MaxSize
-                && ((this.RestrictedSizes != null
-                    && resizeLayer.RestrictedSizes != null
-                    && this.RestrictedSizes.SequenceEqual(resizeLayer.RestrictedSizes))
-                    || (this.RestrictedSizes == resizeLayer.RestrictedSizes))
-                && this.AnchorPoint == resizeLayer.AnchorPoint;
-        }
-
-        /// <summary>
-        /// Returns the hash code for this instance.
-        /// </summary>
+        /// <summary>Indicates whether the current object is equal to another object of the same type.</summary>
+        /// <param name="other">An object to compare with this object.</param>
         /// <returns>
-        /// A 32-bit signed integer that is the hash code for this instance.
-        /// </returns>
+        ///   <see langword="true" /> if the current object is equal to the <paramref name="other" /> parameter; otherwise, <see langword="false" />.</returns>
+        public bool Equals(ResizeLayer other)
+            => other != null
+            && this.Size == other.Size
+            && this.MaxSize == other.MaxSize
+            && this.RestrictedSizes.SequenceEqual(other.RestrictedSizes)
```

Comment:
This can cause an `ArgumentNullException` if `this.RestrictedSizes` or `other.RestrictedSizes` is `null`. Even through the constructor sets this to an empty list, the property is publicly settable and could therefore be set to `null` later.

This would be the simple fix:
```suggestion
            && (this.RestrictedSizes ?? new List<Size>()).SequenceEqual(other.RestrictedSizes ?? new List<Size>())
```

Reply:
Or slightly more complex:
```suggestion
            && (this.RestrictedSizes == null || other.RestrictedSizes == null ? this.RestrictedSizes == other.RestrictedSizes : this.RestrictedSizes.SequenceEqual(other.RestrictedSizes))
```
So if either is `null`, they should both be `null` to be equal, otherwise they're both not `null` and the sequence should be equal 👍